activity.py

```diff
@@ -107,6 +109,12 @@ def build_toolbar(self):
         harder_button.connect('clicked', self._harder_button_cb)
         toolbar_box.toolbar.insert(harder_button, -1)
 
+        self.hole_button = ToggleToolButton('maze-hole')
+        self.hole_button.set_tooltip(_('Add holes'))
```

Comment:
Here you set the tooltip to "Add holes", but the function argument is "add_hole".  Plural or singular should be maintained if possible.

Reply:
Yes, I have changed that.